models/request-headers-1.0.0.yaml

```
@@ -10,68 +62,78 @@ X-Flow-ID:
     (Apple Identifier for mobile Advertising) for iOS, or the
-    [GAID](https://support.google.com/googleplay/android-developer/answer/6048248) (Google mobile Advertising Identifier)
-     for Android.
+    [GAID](https://support.google.com/googleplay/android-developer/answer/6048248)
+    (Google mobile Advertising Identifier) for Android.
   schema:
     type: string
     example: cdda802e-fb9c-47ad-0794d394c912
 
 X-Tenant-ID:
   name: X-Tenant-ID
-  description: |
-    Identifies the tenant initiated the request to the multi tenant Zalando Platform. The X-Tenant-ID must be set
-     according to the Business Partner ID extracted from the OAuth token when a request from a Business Partner
-     hits the Zalando Platform.
+  description: |-
+    The unique identifier of the tenant that initiated the request to the
+    multi-tenant Zalando Platform. The X-Tenant-ID is set to the ID of the
+    business partner extracted from the `OAuth`-token of the request. The
+    `X-Tenant-ID` should be passed-through as generic aspect.
   in: header
   schema:
     type: string
     example: 9f8b3ca3-4be5-436c-a847-9cd55460c495
 
 X-Sales-Channel:
   name: X-Sales-Channel
-  description: |
-    Sales channels are owned by retailers and represent a specific consumer segment being addressed with a
-    specific product assortment that is offered via CFA retailer catalogs to consumers (see fashion platform
-    glossary (internal link))
+  description: |-
+    Unique identifier of the sales channel related to the request. Sales
+    channels are owned by retailers and represent a specific consumer segment
+    being addressed with a specific product assortment that is offered via the
+    CFA retailer catalogs to consumers (see fashion platform glossary). The
+    `X-Sales-Channel` should be passed-through as generic aspect.
   in: header
   schema:
     type: string
     example: 52b96501-0f8d-43e7-82aa-8a96fab134d7
 
 X-Frontend-Type:
   name: X-Frontend-Type
-  description: |
-    Consumer facing applications (CFAs) provide business experience to their customers via different
-    frontend application types, for instance, mobile app or browser. Info should be passed-through as
-    generic aspect — there are diverse concerns, e.g. pushing mobiles with specific coupons, that make use
-    of it. Current range is mobile-app, browser, facebook-app, chat-app, email.
+  description: |-
+    The type of the consumer facing application (CFA) used to initiate the
+    request. CFAs provide business experience to their customers via different
+    frontend application types, `mobile-app`, `browser`, `facebook-app`,
+    `chat-app`, and `email`. The `X-Frontend-Type` should be passed-through as
+    generic aspect - there are diverse concerns, e.g. pushing mobiles with
+    specific coupons, that make use of it.
   in: header
   schema:
     type: string
     example: mobile-app
 
 X-device-Type:
   name: X-device-Type
```

This should be `X-Device-Type`, but changing may break referencing APIs - we need to investigate, whether this is referenced at all.

---

I guess you could change it in `name:`, but leave it as the map key.

---

Yes, I agree, should be possible. Shall I?

---

Changing only the name is compatible and should be done.
Can you simply check, whether `https://opensource.zalando.com/restful-api-guidelines/models/request-headers-1.0.0.yaml#/X-device-Type` is used in active API definitions? If it is not used, I would also change it consistently.

---

I think this should be easy possible by doing a grep over all API specifications. May have some false positives but it at least gives a first idea.

---

The simple search confirmed that this item is referenced at least one API.

---

Discussed in the meeting:
Let's try to keep it compatible by using
```suggestion
  $ref: X-Device-Type

X-Device-Type:
    name: X-device-Type
```